### PR TITLE
added deeplinking as prop

### PR DESCRIPTION
**Pull request by revaret**
### How Has This Been Tested?
1. Clone the swagger-api/swagger-ui repo and made the changes in the code.
2. Execute the `flavors/swagger-ui-react/release/run.sh` script.
3. The compiled library files we need are in the `flavors/swagger-ui-react/dist` folder. Verify the timestamps match when you ran the run.sh script.
4. Copy the files to another folder , ran a example with the generated code.
5. Tested deeplinking by putting the appropriate link in the browser.
6. Repo => https://github.com/revaret/swagger-ui-react

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [ ] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by revaret** [comment]
@tim-lai

File: flavors/swagger-ui-react/index.js [hunk 1]
```diff
@@ -96,11 +97,13 @@ SwaggerUI.propTypes = {
   showMutatedRequest: PropTypes.bool,
   defaultModelExpandDepth: PropTypes.number,
   presets: PropTypes.arrayOf(PropTypes.func),
+  deepLinking:PropTypes.bool
```
Comment (tim-lai) [inline flavors/swagger-ui-react/index.js:100]:
```suggestion
  deepLinking: PropTypes.bool,
```

File: flavors/swagger-ui-react/index.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ export default class SwaggerUI extends React.Component {
       defaultModelExpandDepth: this.props.defaultModelExpandDepth,
       displayOperationId: this.props.displayOperationId,
       showMutatedRequest: typeof this.props.showMutatedRequest === "boolean" ? this.props.showMutatedRequest : true,
+      deepLinking: typeof this.props.deepLinking === "boolean" ? this.props.deepLinking : false
```
Comment (tim-lai) [inline flavors/swagger-ui-react/index.js:26]:
```suggestion
      deepLinking: typeof this.props.deepLinking === "boolean" ? this.props.deepLinking : false,
```

File: flavors/swagger-ui-react/index.js [hunk 3]
```diff
@@ -96,11 +97,13 @@ SwaggerUI.propTypes = {
...
+  deepLinking:PropTypes.bool
 }
 
 SwaggerUI.defaultProps = {
   supportedSubmitMethods: ['get', 'put', 'post', 'delete', 'options', 'head', 'patch', 'trace'],
   docExpansion: "list",
   defaultModelsExpandDepth: 1,
-  presets: []
+  presets: [],
+  deepLinking: false
```
Comment (tim-lai) [inline flavors/swagger-ui-react/index.js:108]:
```suggestion
  deepLinking: false,
```

**Comment by tim-lai** [comment]
@revaret Thanks for the PR! Please fill out the PR template, especially the "How has this been tested" section. Thanks.

**Comment by revaret** [comment]
@tim-lai done

**Comment by tim-lai** [comment]
@revaret PR merged! Thanks for the contribution!

**Pull request merged**